dashboard/modules/reporter/profile_manager.py

```diff
@@ -131,6 +144,125 @@ async def cpu_profile(
         )
         stdout, stderr = await process.communicate()
         if process.returncode != 0:
-            return False, _format_failed_pyspy_command(cmd, stdout, stderr)
+            return False, _format_failed_profiler_command(
+                cmd, self.profiler, stdout, stderr
+            )
         else:
             return True, open(profile_file_path, "rb").read()
+
+
+class MemoryProfilingManager:
+    def __init__(self, profile_dir_path: str):
+        self.profile_dir_path = Path(profile_dir_path) / "memray"
+        self.profile_dir_path.mkdir(exist_ok=True)
+        self.profiler = "memray"
+
+        if not self.profile_dir_path.exists():
+            self.profile_dir_path.mkdir()
+
+    async def get_profile_result(
+        self, pid: int, profiler_filename: str, format="flamegraph", leaks=False
```

Comment:
How difficult is it to support every arg? Also at least native has to be supported